megamek/src/megamek/client/bot/princess/PathRanker.java

```diff
@@ -197,7 +196,7 @@ private List<MovePath> validatePaths(List<MovePath> startingPathList, Game game,
                 // Skip this part if I'm an aero on the ground map, as it's kind of irrelevant
                 // also skip this part if I'm attempting to retreat, as engagement is not the
                 // point here
-                if (!isAirborneAeroOnGroundMap && !getOwner().wantsToFallBack(mover)) {
+                if (!isAirborneAeroOnGroundMap && !getOwner().wantsToFallBack(mover) && !hasNoEnemyAvailable) {
```

Comment:
added here because its useless to look for enemies again if there are no enemies around